fs/config/configfile/configfile.go

```diff
@@ -107,8 +107,27 @@ func (s *Storage) Save() error {
 		return fmt.Errorf("failed to save config file, path is empty")
 	}
 	fs.Debugf(nil, "Saving config file: %v", configPath)
-	dir, name := filepath.Split(configPath)
-	err := file.MkdirAll(dir, os.ModePerm)
+
+	configPathReal := configPath
+	info, err := os.Lstat(configPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("failed to resolve config file path: %w", err)
+		}
+	} else {
+		if info.Mode()&os.ModeSymlink != 0 {
```

Comment:
This is only going to work for absolute symlinks, for relative symlinks it will assume they are relative to rclone's current directory.

Reply:
Right... gotcha! :)